store/rootmulti/store.go

```diff
@@ -761,33 +766,69 @@ func (rs *Store) Snapshot(height uint64, protoWriter protoio.Writer) error {
 		return strings.Compare(stores[i].name, stores[j].name) == -1
 	})
 
+	// CV Load the commit to get actual version ids for each store
+	infos := make(map[string]types.StoreInfo)
+	cInfo := &types.CommitInfo{}
+
+	var err error
+	cInfo, err = getCommitInfo(rs.db, int64(height))
+	if err != nil {
+		return err
+	}
+
+	// convert StoreInfos slice to map
+	for _, storeInfo := range cInfo.StoreInfos {
+		infos[storeInfo.Name] = storeInfo
+	}
+
+	storesKeys := make([]types.StoreKey, 0, len(rs.storesParams))
+	for key := range rs.storesParams {
+		storesKeys = append(storesKeys, key)
+	}
 	// Export each IAVL store. Stores are serialized as a stream of SnapshotItem Protobuf
 	// messages. The first item contains a SnapshotStore with store metadata (i.e. name),
 	// and the following messages contain a SnapshotNode (i.e. an ExportNode). Store changes
 	// are demarcated by new SnapshotStore items.
 	for _, store := range stores {
-		exporter, err := store.Export(int64(height))
+
+		commitID := rs.getCommitID(infos, store.name)
+		storeVersion := commitID.Version
+		rs.logger.Debug("Snapshot Begin", "store", store.name, "version", storeVersion, "hash", hex.EncodeToString(commitID.Hash))
+
+		exporter, err := store.Export(int64(storeVersion))
+		if exporter == nil {
+			rs.logger.Error("Snapshot Failed - exporter nil", "store", store.name)
+			// CV Skip stores that fail to get an exporter
+			//    For example, when iavl/immutable_tree.ndb (nodedb) is nil
+			continue
+		}
 		if err != nil {
+			rs.logger.Error("Snapshot Failed", "store", store.name, "err", err)
 			return err
 		}
 
+		nodeCount := 0
```

Comment:
Why is node count useful?

Reply:
It is helpful to determine a) why a particular store takes a long time to export and b) is a quick way to identify when a store has data (some stores have no data).  

In the case where stores have no data, teams can safely delete the stores in the course of correcting other errors such as a mismatched store height.  Teams often "unintentionally" add stores while copying code and do not know whether transactions have added data.

Reply:
I can see the user case, how does this differ over using iavl viewer or some other tool to check the store?

Reply:
Validators are the primary sources of snapshots.  Validators rarely use iavlviewer to verify their snapshots after creation.  

This is a major problem for example in starname where the snapshot manager was incorrectly configured and creating snapshots with 0 data nodes.

This meant their p2p network was sharing snapshots at various heights all with no data available.

Being able to see that snapshots are being created with an absurd number of nodes upon a binary change is helpful in identifying problems early rather than waiting for a later point in time where snapshot restores and binary changes "seem unrelated".

Reply:
makes sense. Im fine leaving it for the case you mentioned. thanks for the explanation